lib/AST/ParameterPack.cpp

```diff
@@ -186,6 +186,12 @@ bool TypeBase::isParameterPack() {
   while (auto *memberTy = t->getAs<DependentMemberType>())
     t = memberTy->getBase();
 
+  return t->isDirectParameterPack();
+}
+
+bool TypeBase::isDirectParameterPack() {
```

Comment:
Maybe isRootParameterPack()?

Also if you grep for `)->isParameterPack()` you'll find a ton of places you can use your new method, eg in ParameterPack.cpp,

```
      if (expansionLevel == 0 &&
          (isa<PackArchetypeType>(paramType) ||
           (isa<GenericTypeParamType>(paramType) &&
            cast<GenericTypeParamType>(paramType)->isParameterPack()))) {
```